### PR TITLE
Add feature cache oc binary in local folder.

### PR DESCRIPTION
**Pull request by mcarlett**
Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Code is self-descriptive and/or documented

### CODE REVIEW

**Comment by simkam** [comment]
so this is supposed to cache binary between multiple executions? Currently binary is already download only once during execution.

**Comment by mcarlett** [comment]
@simkam yes , this feature can be useful if we use the library in multiple automated test, besides of multiple manual executions in the same host (replace n downloads of the 100+MB with just 1 download)

File: core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java [hunk 1]
```diff
@@ -6,6 +6,9 @@
     public static final String OPENSHIFT_VERSION = "xtf.openshift.version";
     public static final String OPENSHIFT_NAMESPACE = "xtf.openshift.namespace";
     public static final String OPENSHIFT_BINARY_PATH = "xtf.openshift.binary.path";
+    public static final String OPENSHIFT_BINARY_CACHE_ENABLED = "xtf.openshift.binary.cache.enabled";
+    public static final String OPENSHIFT_BINARY_CACHE_PATH = "xtf.openshift.binary.cache.path";
+    public static final String OPENSHIFT_BINARY_CACHE_PATH_DEFAULT_VALUE = "/tmp/xtf/oc-cache";
```
Comment (mnovak1) [inline core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java:11]: I think there could be used more OS agnostic approach and store `oc` in folder like: 
```Files.createTempDirectory(...);```

**Review by mnovak1 — COMMENTED** [review]
I have suggestion for one improvement otherwise LGTM.

File: core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java (hunk 1 above)
Comment (mcarlett) [reply core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java:11]: sure good idea, thanks @mnovak1 I will apply this change

Comment (simkam) [reply core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java:11]: Wouldn't it effectively disable caching with default value? A new temporary directory would be created with each execution, no?

Comment (mcarlett) [reply core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java]: @simkam I update the PR, I cannot create a new temp directory every time, I just used java.io.tmpdir property to retrieve the temp root dir, and I used Paths.get(..) to concatenate root temp and folder

**Review by simkam — APPROVED** [review]

**Review by mnovak1 — APPROVED** [review]

**Pull request merged**